esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2192,6 +2183,75 @@ bool Cws_accessEx::onResourcePermissions(IEspContext &context, IEspResourcePermi
     return true;
 }
 
+bool Cws_accessEx::onResourcePermissionQuery(IEspContext &context, IEspResourcePermissionQueryRequest &req, IEspResourcePermissionQueryResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* ldapSecMgr = queryLDAPSecurityManager(context);
+        if(!ldapSecMgr)
+        {
+            resp.setNoSecMngr(true);
+            return true;
+        }
+
+        checkUser(context, req.getRtype(), req.getRtitle(), SecAccess_Read);
+
+        __int64 pageStartFrom = 0;
+        unsigned pageSize = 100;
+        if (!req.getPageSize_isNull())
+            pageSize = req.getPageSize();
+        if (!req.getPageStartFrom_isNull())
+            pageStartFrom = req.getPageStartFrom();
+
+        ResourcePermissionField sortOrder[2] = {RPFName, RPFterm};
+        CResourcePermissionSortBy sortBy = req.getSortBy();
+        switch (sortBy)
```

Comment:
It's not clear to me why this was done with a switch statement

Reply:
The switch was used in other places due to multiple sortby's. It should not be used here.